hcipy/coronagraphy/vortex.py

```diff
@@ -373,9 +373,22 @@ def backward(self, instance_data, wavefront):
 
 		for i, (jones_matrix, prop) in enumerate(zip(instance_data.jones_matrices, instance_data.props)):
 			if i == 0:
-				pup = Wavefront(prop.backward(wavefront.electric_field))
+				if not wavefront.is_polarized:
+					efield = Wavefront(wavefront.electric_field, input_stokes_vector=(1, 0, 0, 0))
```

Comment:
I mean something like
```python
# Make wavefront polarized if it wasn't already.
if not wavefront.is_polarized:
    wavefront = Wavefront(wavefront.electric_field, input_stokes_vector=(1, 0, 0, 0))

# Propagate backwards through the Fourier filter.
pup = prop.backward(wavefront.electric_field, input_stokes_vector=wavefront.input_stokes_vector)
```
instead of your current if-else statement (so until line 386).

Reply:
yes indeed. This is done now

Reply:
Great! I wasn't 100% sure that worked, so I thought I missed something.